spec/isocalendar.html

```diff
@@ -151,112 +151,146 @@ <h1>Temporal.ISO8601Calendar.prototype.dateDifference ( _one_, _two_, _options_
     </emu-clause>
 
     <emu-clause id="sec-temporal.iso8601calendar.prototype.year">
-      <h1>Temporal.ISO8601Calendar.prototype.year ( _date_ )</h1>
+      <h1>Temporal.ISO8601Calendar.prototype.year ( _dateOrDateTime_ )</h1>
       <p>
-        The `year` method takes one argument _date_.
+        The `year` method takes one argument _dateOrDateTime_.
         The following steps are taken:
       </p>
       <emu-alg>
-        1. <mark>TODO.</mark>
+        1. Perform ? RequireOneOfInternalSlots(_dateOrDateTime_, « [[InitializedTemporalDate]], [[InitializedTemporalDateTime]] »).
```

Comment:
FWIW, I thought it was fine to have RequireInternalSlot(..., [[ISOYear]]) ... since all the types for which this is allowed have an [[ISOYear]] slot. But I think this is fine too. We can always change it if it gets too cumbersome when we add YearMonth, MonthDay, and LocalDateTime.

Reply:
fwiw i don't think we'll want to land a `RequireOneOfInternalSlots` abstract operation in the spec itself - a type should have *all* of the required internal slots, every time.

Reply:
@ljharb I don't understand. These methods are supposed to support several different types.

Reply:
@Ms2ger i'm just saying that in the spec text, we have RequireInternalSlot, but we don't have "RequireInternalSlots" - it's fine that the methods support several different types, but that shouldn't be done by conflating cross-type internal slots. If you want to test for multiple types, then do that with multiple abstract operations.